src/linear_algebra/cross_product.lean

```diff
@@ -0,0 +1,102 @@
+/-
+Copyright (c) 2021 Martin Dvorak. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Martin Dvorak, Kyle Miller, Eric Wieser
+-/
+
+import linear_algebra.bilinear_map
+import linear_algebra.matrix.determinant
+import linear_algebra.vectors
+
+/-!
+# Cross products
+
+This module defines the cross product of vectors in $R^3$ for $R$ a commutative ring.
+
+## Main definitions
+
+* `cross_product` is the cross product of pairs of vectors in $R^3$.
+
+## Main results
+
+* `triple_product_eq_det`
+
+## Notation
+
+The locale `vectors` gives the following notation:
+
+* `×₃` for the cross product
+
+## Tags
+
+crossproduct
+-/
+
+open_locale vectors
+
+variables {R : Type*} [comm_ring R]
+
+/-- The cross product of two vectors in $R^3$ for $R$ a commutative ring. -/
+def cross_product : (fin 3 → R) →ₗ[R] (fin 3 → R) →ₗ[R] (fin 3 → R) :=
+begin
+  apply linear_map.mk₂ R (λ (a b : fin 3 → R),
+    ![ (a 1)*(b 2) - (a 2)*(b 1) ,
+       (a 2)*(b 0) - (a 0)*(b 2) ,
+       (a 0)*(b 1) - (a 1)*(b 0) ]);
+  intros;
+  simp only [vec3_add,
+    pi.add_apply, algebra.id.smul_eq_mul, matrix.smul_cons, matrix.smul_empty, pi.smul_apply];
+  apply vec3_eq;
+  ring,
+end
+
+localized "infixl ` ×₃ `: 68 := cross_product" in vectors
+
+lemma cross_product_def (a b : fin 3 → R) :
+  a ×₃ b =
+    ![ (a 1)*(b 2) - (a 2)*(b 1) ,
+       (a 2)*(b 0) - (a 0)*(b 2) ,
+       (a 0)*(b 1) - (a 1)*(b 0) ] :=
+rfl
```

Comment:
These parens are redundant:
```suggestion
lemma cross_product_apply (a b : fin 3 → R) :
  a ×₃ b = ![a 1 * b 2 - a 2 * b 1,
             a 2 * b 0 - a 0 * b 2,
             a 0 * b 1 - a 1 * b 0] :=
rfl
```
(note also that `_apply` is the usual suffix for this type of thing)

Reply:
I'd like to apply only the second change (replacing `cross_product_def` by `cross_product_apply`).